### PR TITLE
Enhancement: Update phpspec/phpspec

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] updates `phpspec/phpspec`

💁‍♂️ For reference, see https://github.com/phpspec/phpspec/compare/2.3.0...3.3.0.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -36,8 +36,8 @@
     "codeclimate/php-test-reporter": "0.2.0",
     "fzaninotto/faker": "^1.5.0",
     "humbug/humbug": "^1.0.0-alpha1",
-    "phpspec/nyan-formatters": "~1.0",
-    "phpspec/phpspec": "^2.4.1",
+    "phpspec/nyan-formatters": "^2.0.0",
```
Comment (localheinz) [inline composer.json]: Updating this as well, previously installed version is not compatible.

File: phpspec.yml [hunk 2]
```diff
@@ -4,6 +4,6 @@ suites:
     psr4_prefix: Refinery29\Piston
 
 extensions:
-  - PhpSpec\NyanFormattersExtension\Extension
+  PhpSpec\NyanFormattersExtension\Extension: ~
```
Comment (localheinz) [inline phpspec.yml]: Configuration has slightly changed, see http://www.phpspec.net/en/stable/cookbook/extensions.html#configuration.

File: composer.json (hunk 1 above)
Comment (rcatlin) [inline composer.json]: This is probably the most critical dependency we have 🙌

**Comment by localheinz** [comment]
Thank you, @rcatlin!

**Pull request merged**